### PR TITLE
Jones/validate email

### PR DESCRIPTION
**Pull request by taylor-jones**
1) Adds live validation to initial user creation form where:
 - available usernames and emails are checked on the fly.
 - Valid email addresses are required
 - Matching password and re-password are required.

2) Fixes an issue in the UserAccountService with logging the creation of a new user _by_ the new user, where it avoids any issues w/ integrity constraints.

### CODE REVIEW

File: erp-client/src/app/components/createAccount/set-recovery-questions/set-recovery-questions.component.html [hunk 1]
```diff
@@ -29,23 +29,6 @@
...
-<!--              </td>-->
-<!--            </tr>-->
-<!--            <tr>-->
-<!--              <td>-->
-<!--                <mat-form-field class="set-password">-->
-<!--                  <input matInput placeholder="Re-Enter Password" type="password" [(ngModel)]="rePassword" name="rePassword" required>-->
-<!--                </mat-form-field>-->
-<!--              </td>-->
-<!--            </tr>-->
-<!--          </div>-->
```
Comment (tmrietz) [inline erp-client/src/app/components/createAccount/set-recovery-questions/set-recovery-questions.component.html:48]: 💯

File: erp-server/src/main/java/com/ttt/erp/controller/UserAccountController.java [hunk 2]
```diff
@@ -131,6 +131,6 @@ public UserAccount getUserAccount(@PathVariable("username") final String usernam
 
     @PostMapping("/newAccount")
     public Optional<UserAccount> addUserCreatedAccount (@RequestBody UserAccount newUserAccount) {
-        return this.service.createUser(0L, newUserAccount);
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/controller/UserAccountController.java]: Yep, nice 👍

File: erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java [hunk 3]
```diff
@@ -41,25 +41,23 @@ public UserAccount userFromPrincipal(Principal principal) {
         return Optional.ofNullable(this.repository.findByUsername(username).getSignature());
     }
 
-    public Optional<UserAccount> createUser(Long userAccountId, UserAccount user) {
+    public Optional<UserAccount> createUser(UserAccount user) {
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java]: This also gets called on admin user creates. Was the function signature changed for the controller mapping to that endpoint?

Comment (taylor-jones) [reply erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java]: No, I didn't realize the admin user creation used this too. I'll check on it.

Comment (taylor-jones) [reply erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java]: Wait, actually -- where do I need to look for that?
Edit: I think I see what needs to be fixed. I'm working on the fix.

Comment (tmrietz) [reply erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java]: I don't think we can do this. The userAccountId param is supposed to be for the actingUser (the one submitting the request on the client). By removing it we'll never log any user creations by the admin user. Sorry, I should have named the param something better like actingUser.

The call by an admin's in UserAccountController.addUserAccount . If you're in VSCode you can do Edit -> Find in Files -> createUser(

**Comment by taylor-jones** [comment]
It should all be good now for creating admin or non-admin users. I've tested both.

File: erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java (hunk 3 above)
Comment (taylor-jones) [reply erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java]: Yeah you were right. It needed to be put back in place and handled differently.

**Review by tmrietz — APPROVED** [review]
Sweet, thanks again. Let's merge and be code-complete!

**Pull request merged**